puppeteer-pdf.js

```diff
@@ -1,130 +1,111 @@
+  "Paper width, accepts values labeled with units."
+).option(
+  "-h, --heigh [height]",
+  "Paper height, accepts values labeled with units."
+).option(
+  "-mt, --marginTop [margin]",
+  "Top margin, accepts values labeled with units."
+).option(
+  "-mr, --marginRight [margin]",
+  "Right margin, accepts values labeled with units."
+).option(
+  "-mb, --marginBottom [margin]",
+  "Bottom margin, accepts values labeled with units."
+).option(
+  "-ml, --marginLeft [margin]",
+  "Left margin, accepts values labeled with units."
+).option("-d, --debug", "Output Puppeteer PDF options").option(
+  "-wu, --waitUntil [choice]",
+  "waitUntil accepts choices load, domcontentloaded, networkidle0, networkidle2. Defaults to 'networkidle2'.",
+  "networkidle2"
+).option("-oBg, --omitBackground", "Omits background", false).parse(process.argv);
+var cli_default = program;
 
-(async () => {
-  let options = {};
-
-  // Loop through options
-  _.each(cli.options, function(option) {
-    const optionName = option.name();
-    if (!_.isNil(cli[optionName]) && !["version"].includes(optionName)) {
-      const optionValue = cli[optionName];
-
-      if (_.startsWith(optionName, "margin")) {
-        // Margins need to be combined into an object
-        _.set(
-          options,
-          ["margin", optionName.replace("margin", "").toLowerCase()],
-          optionValue
-        );
-      } else {
-        _.set(options, optionName, optionValue);
+// src/options.js
+var fs = __require("fs");
+var prepareOptions = (optionsFromCLI) => Object.keys(optionsFromCLI).reduce((acc, optionKey) => {
+  if (optionKey == "version" || typeof optionKey == "object") {
+    return acc;
+  }
+  const value = optionsFromCLI[optionKey];
+  if (optionKey.startsWith("margin")) {
+    const key = optionKey.replace("margin", "");
+    return {
+      ...acc,
+      margin: {
+        ...acc.margin,
+        [key]: value
       }
-    }
-  });
-
-  // Check if we need to read header or footer templates from files
-  _.each(["headerTemplate", "footerTemplate"], function(template) {
-    if (_.get(options, template, "").startsWith("file://")) {
-      options[template] = fs.readFileSync(
-        options[template].replace("file://", ""),
-        "utf-8"
-      );
-    }
-  });
+    };
+  }
+  if (["headerTemplate", "footerTemplate"].includes(optionKey) && value.startsWith("file://")) {
+    const loadedFile = fs.readFileSync(
+      value.replace("file://", ""),
+      "utf-8"
+    );
+    return {
+      ...acc,
+      [optionKey]: loadedFile
+    };
+  }
+  return {
+    ...acc,
+    [optionKey]: value
+  };
+}, {});
 
-  const browser = await puppeteer.launch({ args: ["--no-sandbox"] });
+// main.js
+(async () => {
+  const cliOptions = cli_default.opts();
+  const options = prepareOptions(cliOptions);
+  const executablePath = process.env.CHROME_BIN || "/usr/bin/google-chrome-stable";
+  const browser = await puppeteer.launch({ args: ["--no-sandbox"], executablePath });
```

Comment:
Asking because I don't know - why do we need to have this here too when we have the (apparently) same code in main.js ?

Reply:
This file is run on docker file as the main.js is the entry point for the development :D